test/save_load_scfres.jl

```diff
@@ -0,0 +1,37 @@
+# This test is based on example examples/collinear_magnetism.jl
+# Import statements are not all included in front to test for 
+# error statements in dispatch
+
+using DFTK
+
+a = 5.42352
+lattice = a / 2 * [[-1  1  1];
+                   [ 1 -1  1];
+                   [ 1  1 -1]]
+Fe = ElementPsp(:Fe, psp=load_psp("hgh/lda/Fe-q8.hgh"))
+atoms = [Fe => [zeros(3)]];
+kgrid = [3, 3, 3]
+Ecut = 15
+magnetic_moments = [Fe => [4, ]];
+model = model_LDA(lattice, atoms, magnetic_moments=magnetic_moments, temperature=0.01)
+basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid)
+ρspin = guess_spin_density(basis, magnetic_moments)
+scfres = self_consistent_field(basis, tol=1e-6, ρspin=ρspin, mixing=KerkerMixing());
+
+@test_throws ErrorException save_scfres("MyVTKfile.vts",scfres)
+@test_throws ErrorException save_scfres("MyVTKfile.jld",scfres)
```

Comment:
Some other tests actually load JLD2 and you should not rely on this being the only use of WriteVTK either (tests should be decoupled).

Reply:
What exactly do we mean by decouple here? A test should run different from runtests.jl ? Or can we just shift this test to the first( this can cause problems in future I guess.)

Reply:
Yes combining the tests is fine, but this won't help. The problem is that you rely that the tests are executed in a particular order (because no test may load `JLD2` before this test runs, otherwise it will fail ... try it!), but there is no guarantee the test order will stay as it is in the future.

Reply:
`save_load_scfres` will run a completely separate test? I am a little hesitant to do that because ideally `test/runtests.jl` should run all tests in the package. We can use a package [Safetests.jl](https://github.com/YingboMa/SafeTestsets.jl) as mentioned in a [discourse thread](https://discourse.julialang.org/t/best-practices-for-julia-unit-testing/30858/3) but then I would not bother adding another package for one error test which can be done separately. Another thing we can do is add a `@testset` before `DFTK`, but that will be a major design issue. I am confused.

Reply:
Yes, sorry to be confusing. Indeed `runtests.jl` should run all tests. But it could happen that in future restructuring of the test suite we change the order of the includes in `runtests.jl`. The problem is that your test assumes that e.g. `JLD2` has not yet been imported with a `using` statement in any of the testcases running before your testcase. That might not be true in the future when we might restructure the test suite, so you should not make this assumption in the test.

Reply:
And I would not add Safetests.jl as another dependency actually. I don't think it's needed.

Reply:
On another note I feel the test for errors for non loaded modules are redundant. If they are not loaded there will be obvious errors and `@test_throws` does not test for which error whether it is MethodError or any user defined Error. Therefore the first two lines serves no purpose as of such and rest two tests can be integrated easily in a normal structure.

Reply:
I agree. I think the way you have it right now is completely fine.